### PR TITLE
Update to latest kubernetes version supported & update CAPZ to latest version

### PR DESCRIPTION
**Pull request by nellyk**

### CODE REVIEW

File: clusterctl.env [hunk 1]
```diff
@@ -2,6 +2,8 @@
 export EXP_AKS=true
 export EXP_MACHINE_POOL=true
 export EXP_CLUSTER_RESOURCE_SET=false
+export EXP_AKS_RESOURCE_HEALTH=true
```
Comment (dtzar) [inline clusterctl.env]: Let's leave this out or at least set to false.

File: charts/azure-managed-cluster/values.yaml [hunk 2]
```diff
@@ -47,14 +49,14 @@ controlplane:
...
-   sku: Standard_B4ms
-   nodecount: 2
+   sku: Standard_D4d_v5
+   nodecount: 5
    osDiskSizeGB: 100
  - name: agentpool1
    mode: User
    nodecount: 2
-   sku: Standard_B4ms
+   sku: Standard_D4d_v5
```
Comment (dtzar) [inline charts/azure-managed-cluster/values.yaml:146]: Reason why this size?  I'd like to leave a smaller size for demo/cost purposes for default.  Above, we're putting in Standard_B2s.  These should all match.

File: charts/azure-managed-cluster/values.yaml [hunk 3]
```diff
@@ -6,13 +6,12 @@
...
 ## The subscription ID
 subscriptionID: ""
-
 identity:
   clientId: ""
   tenantId: ""
   clusterIdentityName: "cluster-identity"
   clusterIdentitySecretName: "cluster-identity-secret"
-  clusterIdentitySecretNamespace: "default"
+  clusterIdentitySecretNamespace: "default3"
```
Comment (dtzar) [inline charts/azure-managed-cluster/values.yaml]: why default3?

File: README.md [hunk 4]
```diff
@@ -80,7 +80,7 @@ kubectl create secret generic "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" --from-lit
 Initialize Cluster API and install Azure CAPZ provider version 0.5.3(uses alphav4 capi)
```
Comment (dtzar) [inline README.md]: Fix langugage here

File: charts/azure-managed-cluster/values.yaml (hunk 3 above)
Comment (nellyk) [reply charts/azure-managed-cluster/values.yaml]: this was a test, i'll change it back

File: charts/azure-managed-cluster/values.yaml (hunk 2 above)
Comment (nellyk) [reply charts/azure-managed-cluster/values.yaml:146]: i'll update this to cover production economy

**Review by dtzar — APPROVED** [review]

File: charts/azure-managed-cluster/values.yaml (hunk 2 above)
Comment (ams0) [reply charts/azure-managed-cluster/values.yaml:146]: I always use B's for my clusters, but anyway, this can be user-set

**Review by dtzar — APPROVED** [review]

**Pull request merged**